examples/flutter_gallery/lib/demo/customized/home.dart

```diff
@@ -0,0 +1,836 @@
+      ).animate(
+        CurvedAnimation(
+          curve: Curves.fastOutSlowIn,
+          parent: _numberCounterAnimationController,
+        ),
+      );
+    });
+    _numberCounterAnimationController.forward(from: 0.0);
+  }
+
+  Timer _animateRunCounter() {
+    const Duration duration = Duration(milliseconds: 300);
+    return Timer(duration, _updateRunCounter);
+  }
+
+  GestureDetector _buildAppBar() {
+    return GestureDetector(
+      onTap: () {
+        double screenHeight = MediaQuery.of(context).size.height -
+            MediaQuery.of(context).padding.top;
+        if (Theme.of(context).platform == TargetPlatform.iOS) {
+          screenHeight -= 70.0;
+        }
+        final double halfScreen = screenHeight * 0.5;
+        double scrollToOffset =
+            _scrollController.offset <= halfScreen ? 0.0 : screenHeight;
+        if (_scrollController.offset == 0) {
+          scrollToOffset = screenHeight;
+        } else if (_isStatsBoxFullScreen) {
+          scrollToOffset = 0.0;
+        }
+        _scrollController.animateTo(
+          scrollToOffset,
+          curve: Curves.easeOut,
+          duration: const Duration(milliseconds: 300),
+        );
+      },
+      child: Container(
+        color: const Color(0xFF212024),
+        height: 70.0,
+        child: Stack(
+          children: <Widget>[
+            const Positioned(
+              left: 26.0,
+              top: 0.0,
+              bottom: 0.0,
+              child: Center(
+                child: Text(
+                  'VIEW MY STATS',
+                  style: TextStyle(
+                    color: Color(0xFF02CEA1),
+                    fontSize: 16.0,
+                    fontWeight: FontWeight.w900,
+                  ),
+                ),
+              ),
+            ),
+            Positioned(
+              right: 20.0,
+              top: 0.0,
+              bottom: 0.0,
+              child: RotationTransition(
+                turns: _rotationAnimation,
+                child: const RotatedBox(
+                  quarterTurns: 2,
+                  child: ImageIcon(
+                    AssetImage('assets/images/customized/ic_circle_arrow.png'),
+                    color: Colors.white,
+                  ),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _buildBackButton() {
+    final TargetPlatform platform = Theme.of(context).platform;
+    final IconData backIcon = platform == TargetPlatform.android
+        ? Icons.arrow_back
+        : Icons.arrow_back_ios;
+    return Container(
+      height: 70.0,
+      width: 70.0,
+      child: Material(
+        color: const Color(0x00FFFFFF),
+        child: IconButton(
+          icon: Icon(backIcon, color: Colors.white),
+          onPressed: () {
+            Navigator.of(context).pop();
+          },
+        ),
+      ),
+    );
+  }
+
+  Widget _buildBody() {
+    return Opacity(
+      opacity: _statsOpacity,
```

Comment:
If you change this value and call setState on this State object, everything will rebuild. If you have a widget that takes a specific widget instance and listens to a listenable (such as via ValueListenableBuilder), then only one node in the tree will rebuild. 

I only read up until here so I don't know if you need to rebuild everything.

Reply:
After refactoring we don't need to rebuild. We also cache the main content widget.